app/src/main/java/com/wireguard/android/services/NotificationService.kt

```diff
@@ -0,0 +1,84 @@
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.wireguard.android.R
+import com.wireguard.android.activity.MainActivity
+import com.wireguard.android.model.TunnelManager
+import timber.log.Timber
+import java.text.SimpleDateFormat
+import java.util.*
+
+class NotificationService: IntentService("Notification Service") {
+
+    private var name: String? = null
+    private var id: Int? = null
+    private val handler = Handler()
+    private val timer = Timer()
+    private lateinit var notificationManager: NotificationManagerCompat
+    private lateinit var builder: NotificationCompat.Builder
+
+
+    override fun onHandleIntent(intent: Intent?) {
+        id = intent?.getIntExtra("id", 0)
+        name = intent?.getStringExtra("name")
+        notificationManager = NotificationManagerCompat.from(this)
+        builder = createNotificationBuilder(name!!)
+        startForeground(id!!, builder.build())
+        updateNotifications()
+    }
+
+    private fun updateNotifications() {
+        val builder = createNotificationBuilder(name!!)
+        val dateFormat = SimpleDateFormat("HH:mm:ss", Locale.getDefault())
+        dateFormat.timeZone = TimeZone.getTimeZone("UTC")
+        val startTime = System.currentTimeMillis()
+
+        timer.scheduleAtFixedRate(object: TimerTask() {
+            override fun run() {
+                handler.post {
+                    val elapsed = System.currentTimeMillis() - startTime
+                    Timber.d(elapsed.toString())
+                    builder.setContentTitle("$name | ${dateFormat.format(Date(elapsed))}")
+                    notificationManager.notify(id!!, builder.build())
+                }
+            }
+        }, 0, 1000)
+    }
+
+    private fun createNotificationBuilder(tunnelName: String) : NotificationCompat.Builder {
+        val intent = Intent(this, MainActivity::class.java)
+        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TASK or Intent.FLAG_ACTIVITY_NEW_TASK
+        val pendingIntent = PendingIntent.getActivity(this, 0, intent, 0)
+        val builder = NotificationCompat.Builder(
+                this,
+                TunnelManager.NOTIFICATION_CHANNEL_ID
+        )
+        builder.setContentTitle(this.getString(R.string.notification_channel_wgquick_title))
+                .setContentText(tunnelName)
+                .setContentIntent(pendingIntent)
+                .setOngoing(true)
+                .setOnlyAlertOnce(true)
+                .setPriority(Notification.FLAG_ONGOING_EVENT)
+                .setSmallIcon(R.drawable.ic_qs_tile)
+        return builder
+    }
+
+    override fun stopService(name: Intent?): Boolean {
+        timer.cancel()
+        notificationManager.cancel(id!!)
+        return super.stopService(name)
+    }
+
+    override fun onDestroy() {
+        Timber.d("ondestroy called")
+        timer.cancel()
+        notificationManager.cancel(id!!)
+        super.onDestroy()
+    }
+}
```

Comment:
Needs a trailing newline. Read the [contributing guidelines](https://github.com/msfjarvis/viscerion/blob/master/CONTRIBUTING.md) about setting up the `pre-push-recommended` hook to avoid style violations.